lib/veri_traci.ml

```diff
@@ -0,0 +1,89 @@
+open Bap.Std
+module Bap_value = Value
+open Bap_traces.Std
+open Monad.State
+open Bil.Result
+
+type event = Trace.event
+
+let mem_of_arch arch = 
+  let (module T : Target) = target_of_arch arch in
+  T.CPU.mem
+
+(** TODO: what is a behaviour of update_event method?  *)
+class context arch = object(self:'s)
+  inherit Bili.context
+  val mem  : var = mem_of_arch arch
+  val arch : arch = arch
+  method arch = arch
+  method mem = mem
+  method update_event: event -> 's = fun ev -> failwith "unimplemented"
+end
+
+let mem_operands mv = 
+  Size.of_int_opt (Bitvector.bitwidth (Move.data mv)) |> function
+  | None -> None
+  | Some sz ->
+    let addr = Bil.int (Move.cell mv) in
+    let data = Bil.int (Move.data mv) in
+    Some (addr, data, sz)
+
+class ['a] t = object(self)
+  constraint 'a = #context
+  inherit ['a] Bili.t
+
+  method private eval_reg: var move -> 'a u = fun mv ->  
+    self#eval_move (Move.cell mv) (Bil.int (Move.data mv))
+
+  method private eval_mem_op op mv : 'a r =
+    match mem_operands mv with 
+    | None -> 
```

Comment:
I think it is better to raise an exception here. This is a point for discussion, but returning bot here is a bad idea.